app/helpers/pages_helper.rb

```diff
@@ -1,2 +1,25 @@
 module PagesHelper
+  # Determine if child is a subpath of parent
+  #
+  # If the relative path from the child to the supposed parent includes '..'
+  # then child is not a subpath of parent
+  #
+  # @param parent [Pathname]
+  # @param child [Pathname]
+  # @return Boolean
+  def child?(parent, child)
+    ! child.expand_path.relative_path_from(
+      parent.expand_path
+    ).each_filename.to_a.include?(
+      '..'
+    )
+  end
+
+  def whitelist
+    ::Rails.application.config.whitelist_paths
+  end
+
+  def is_path_whitelisted?
+    whitelist.empty? || whitelist.any?{|parent| child?(parent, @pathname)}
+  end
```

Comment:
The `helpers` directory in Rails is meant for view helpers, instead of just general purpose helpers. That is why when you added the helpers they did not appear available in the controller until you explicitly included the methods.

Normally these type of methods would be added to a dedicated configuration object (such as our domain specific ConfigurationSingleton's in other apps) or a domain model, or even a policy object (https://codeclimate.com/blog/7-ways-to-decompose-fat-activerecord-models/). In this case since that controller is small, there is no other server side code and the controller has no private methods, you could just add these as private or protected methods to the controller itself and save time for refactoring after we factor in Tyler's work.